Cargo.toml

```diff
@@ -59,6 +59,7 @@ thiserror = "1.0.14"
 anyhow = "1.0.28"
 async-trait = "0.1.31"
 url = "2.1.1"
+trust-dns-resolver = "0.19.5"
```

Comment:
if we use an external dns resolver it should be non blocking and integrated with the existing async-std runtime, so likely https://github.com/bluejekyll/trust-dns/tree/master/crates/async-std-resolver

Reply:
Refactored to be async.

It seems CircleCI needs fixing. Both jobs are failing with _Permission denied_.